Source/Core/Common/ChunkFile.h

```diff
@@ -35,20 +35,8 @@
 
 // ewww
 
-#ifndef __has_feature
-#define __has_feature(x) (0)
-#endif
-
-#if (__has_feature(is_trivially_copyable) &&                                                       \
-     (defined(_LIBCPP_VERSION) || defined(__GLIBCXX__))) ||                                        \
-    (defined(__GNUC__) && __GNUC__ >= 5) || defined(_MSC_VER)
-#define IsTriviallyCopyable(T)                                                                     \
-  std::is_trivially_copyable<typename std::remove_volatile<T>::type>::value
-#elif __GNUC__
-#define IsTriviallyCopyable(T) std::has_trivial_copy_constructor<T>::value
-#else
-#error No version of is_trivially_copyable
-#endif
+template <typename T>
+constexpr bool IsTriviallyCopyable = std::is_trivially_copyable<std::remove_volatile_t<T>>::value;
```

Comment:
This should probably have a comment indicating this should be removed and replaced with just `std::is_trivially_copyable<T>` in the future, since it can allow invoking undefined behavior (`volatile` types are inherently non-trivially-copyable, so this ignores the type qualifier). This is just a silly workaround for the fact we use `volatile` on things we push out to the save state file, when we should really be using proper atomics.

Reply:
Done